sql/core/src/test/scala/org/apache/spark/sql/DataFrameComplexTypeSuite.scala

```diff
@@ -26,6 +26,38 @@ import org.apache.spark.sql.test.SharedSQLContext
 class DataFrameComplexTypeSuite extends QueryTest with SharedSQLContext {
   import testImplicits._
 
+  test("primitive type on array") {
+    val df = sparkContext.parallelize(Seq(1, 2), 1).toDF("v")
```

Comment:
I added unit tests to check whether the ```zeroOutNullBytes``` is eliminated.
I eliminated three cases such as ```array on array```. I think that we need to unit tests to generate an array, struct, or map, in order to check it works well even when we eliminate ```zeroOutNullBytes```.